bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java

```diff
@@ -0,0 +1,550 @@
+                } else {
+                    return new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "HTTP returncode " + response.getStatus());
+                }
+            } else {
+                logger.debug("Authenticating: Response was null");
+                return new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed retrieving a response from the server.");
+            }
+        }
+        return new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                "Cannot build request from credentials.");
+    }
+
+    protected String invokeAndParse(String vehicleId, String command, String payLoad, WebTarget target) {
+        logger.debug("Invoking: {}", command);
+
+        if (vehicleId != null) {
+            Response response;
+
+            if (payLoad != null) {
+                if (command != null) {
+                    response = target.resolveTemplate("cmd", command).resolveTemplate("vid", vehicleId).request()
+                            .header("Authorization", "Bearer " + logonToken.access_token)
+                            .post(Entity.entity(payLoad, MediaType.APPLICATION_JSON_TYPE));
+                } else {
+                    response = target.resolveTemplate("vid", vehicleId).request()
+                            .header("Authorization", "Bearer " + logonToken.access_token)
+                            .post(Entity.entity(payLoad, MediaType.APPLICATION_JSON_TYPE));
+                }
+            } else {
+                if (command != null) {
+                    response = target.resolveTemplate("cmd", command).resolveTemplate("vid", vehicleId)
+                            .request(MediaType.APPLICATION_JSON_TYPE)
+                            .header("Authorization", "Bearer " + logonToken.access_token).get();
+                } else {
+                    response = target.resolveTemplate("vid", vehicleId).request(MediaType.APPLICATION_JSON_TYPE)
+                            .header("Authorization", "Bearer " + logonToken.access_token).get();
+                }
+            }
+
+            JsonParser parser = new JsonParser();
+
+            if (!checkResponse(response, false)) {
+                logger.debug("An error occurred while communicating with the vehicle during request {}: {}:{}",
+                        new Object[] { command, (response != null) ? response.getStatus() : "",
+                                (response != null) ? response.getStatusInfo() : "No Response" });
+                return null;
+            }
+
+            try {
+                JsonObject jsonObject = parser.parse(response.readEntity(String.class)).getAsJsonObject();
+                logger.trace("Request : {}:{}:{} yields {}",
+                        new Object[] { command, payLoad, target.toString(), jsonObject.get("response").toString() });
```

Comment:
See above and

You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.